packages/xchain-ethereum/src/client.ts

```diff
@@ -57,16 +48,18 @@ import {
 import { getGasOracle } from './etherscan-api'
 
 const ethAddress = '0x0000000000000000000000000000000000000000'
+const MAX_UINT = 2 ** 256 - 1
 
 /**
  * Interface for custom Ethereum client
  */
 export interface EthereumClient {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   call<T>(asset: Address, abi: ethers.ContractInterface, func: string, params: Array<any>): Promise<T>
-  vaultTx(opts: VaultTxOpts): Promise<TransactionResponse>
-  normalTx(opts: NormalTxOpts): Promise<TransactionResponse>
-  erc20Tx(opts: Erc20TxOpts): Promise<TransactionResponse>
+
+  isApproved(spender: Address, sender: Address, amount: BaseAmount): Promise<boolean>
+  approve(spender: Address, sender: Address, amount?: BaseAmount): Promise<TxHash>
+  erc20Transfer(opts: Erc20TxOpts): Promise<TxHash>
 
   estimateGasNormalTx(params: NormalTxOpts): Promise<BaseAmount>
   estimateGasERC20Tx(params: Erc20TxOpts): Promise<BaseAmount>
```

Comment:
Do not have this, have a single `estimateGas(opts)` and return the correct gas amount whether ETH or ERC20